core/custom_field_api.php

```diff
@@ -571,6 +571,16 @@ function custom_field_update( $p_field_id, array $p_def_array ) {
 		trigger_error( ERROR_CUSTOM_FIELD_NAME_NOT_UNIQUE, ERROR );
 	}
 
+  # if foreground color is set to default (black), do not store it
+  if ( $p_def_array['color_foreground'] == '#000000') {
```

Comment:
Do not use magic strings to find out if a value is set or not.
See comment above concerning NULL values.
Keep also in mind that black/white are not always the default colors.

Reply:
removed input type color. set to plain text input.
therefore, magic strings got obsolete and were also removed.